src/CheckersBoard.java

```diff
@@ -1,553 +1,403 @@
+      	if (!correctOwner(fromI, fromJ)) {
+	    //	    throw new CheckersIllegalMoveException("Thats not your piece");
+	    System.out.println("Thats not your piece");
+	    error = true;
+	    }
+	else if (toI > 7 || toJ > 7) { //check if out of bounds
+	    // throw new CheckersIllegalMoveException("You cant move off the board");
+	    System.out.println("You cant move off the board");
+	    error = true;
+	}
+	else if (toI == x1 && toJ == y1) { // tired to combine all this into one if statement with ||
+	    squareOpen(toI, toJ);          // but wasnt working and worked when separated 
+	}
+	else if (toI == x2 && toJ == y2) {
+	    squareOpen(toI, toJ);
+	}
+	else if (toI == x3 && toJ == y3) {
+	    squareOpen(toI, toJ);
+	}
+	else if (toI == x4 && toJ == y4) {
+	    squareOpen(toI, toJ);
+	}
+	else if (toI == jumpULI && toJ == jumpULJ) { //see comment above
+	    checkJump(fromI, fromJ, toI, toJ);
+	}
+	else if (toI == jumpURI && toJ == jumpURJ) {
+	    checkJump(fromI, fromJ, toI, toJ);
+	}
+	else if (toI == jumpDLI && toJ == jumpDLJ) {
+	    checkJump(fromI, fromJ, toI, toJ);
+	}
+	else if (toI == jumpDRI && toJ == jumpDRJ) {
+	    checkJump(fromI, fromJ, toI, toJ);
+	}
+	else {
+	    validMove = false;
+	}
+    }
 
-	     // S isn't 2 chars, invalid, return entry
-	    if (r.length() != 2) return s;
+    public void squareOpen(int i, int j) {
+        if (pieces[i][j] == ' ') {
+	    validMove = true;
+	}
+	else {
+	    validMove = false;
+	}
+    }
+    
+    public void jumped(int i, int j) {
+	if (turn == 'x') {
+	    if (pieces[i][j] == 'o') {
+		validMove = true;
+	    }
+	}
+	else if (turn == 'o') {
+	    if (pieces[i][j] == 'x') {
+		validMove = true;
+	    }
+	}
+    }
+    public void checkJump(int fromI, int fromJ, int toI, int toJ) {
+	if (toI == jumpULI && toJ == jumpULJ) {
+	    jumped(x1,y1);
+	    if (validMove = true) {
+		canJump = true;
+		jumpedI = x1;
+		jumpedJ = y1;
+	    }
+	}
+	else if (toI == jumpURI && toJ == jumpURJ) {
+	    jumped(x2, y2);
+	    if (validMove = true) {
+                canJump	= true;
+		jumpedI = x2;
+		jumpedJ = y2;
+	    }
+	}
+	else if (toI == jumpDLI && toJ == jumpDLJ) {
+	    jumped(x3, y3);
+	    if (validMove = true) {
+                canJump	= true;
+		jumpedI = x3;
+		jumpedJ = y3;
+	    }
+	}
+	else if (toI == jumpDRI && toJ == jumpDRJ) {
+	    jumped(x4, y4);
+	    if (validMove = true) {
+                canJump	= true;
+		jumpedI = x4;
+		jumpedJ = y4;
+	    }
+	}
+	else {
+	    validMove = false;
+	    canJump = false;
+	}
 
-	    
-	    Cpre = r.charAt(0);
-	    Csuf = r.charAt(1);
-	    
-	    // First make sure that format is <Letter><Digit>
-	    if (Character.isDigit(Cpre) && Character.isLetter(Csuf)) { // Format is <Digit><Letter>, switch it
-		char temp = Csuf;
-		Csuf = Cpre;
-		Cpre = temp;
+    }
+    
+    public Boolean kingCheck(int i, int j){
+	if (turn == 'x') {
+	    if (i == 0 && (j == 1 || j == 3 || j == 5 || j == 7)){
```

Comment:
Does the check on `j` make sense here?  I assume you prevent people from moving to invalid squares already...